db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,145 @@
+        # Patient records updated here are solely updated (no History Items included)
+        Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+        Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+      end
+      custom_translations.each do |key, value|
+        key = key.to_s
+        if value === 'bilingual english/spanish'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'eng', secondary_language: 'spa' })
+        elsif value === 'bilingual spanish/english'
+          Patient.where(purged: false).where(primary_language: key).update_all({ primary_language: 'spa', secondary_language: 'eng' })
+        elsif TRANSLATION_COMMENTS.include?(key)
+          note = "Primary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s primary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id, created_by, comment, history_type, created_at, updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND primary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+
+          note = "Secondary language was listed as '#{key}' \
+                which could not be matched to the standard language list. This monitoree’s secondary \
+                language has been updated to '#{value.nil? ? 'blank' : Languages.all_languages[value.to_sym][:display]}'. You \
+                may update this value."
+          execute <<-SQL.squish
+                INSERT INTO histories (patient_id, created_by, comment, history_type, created_at, updated_at)
+                SELECT id, 'Sara Alert System', "#{note}", 'System Note', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP
+                FROM patients
+                WHERE purged = false AND secondary_language = "#{key}"
+          SQL
+          Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+        end
+      end
+    end
+  end
+
+  def down
+    execute <<-SQL.squish
+        UPDATE patients SET primary_language = legacy_primary_language, secondary_language = legacy_secondary_language
+    SQL
+
+    remove_column :patients, :legacy_primary_language
+    remove_column :patients, :legacy_secondary_language
+
+    # Destroy all System Notes that contain the text `language was listed as`
+    History.where('history_type = ? AND comment like ?', 'System Note', '%language was listed as%').destroy_all
+  end
+
+  def match_language(lang)
```

Comment:
Is this method doing the same thing as [`Languages.normalize_and_get_language_name`](https://github.com/SaraAlert/SaraAlert/blob/1344-support-all-standard-languages/app/lib/languages.rb#L37)? Unless I'm missing something, can we just use that instead?

Reply:
Done.